src/frontend/org/voltdb/NativeLibraryLoader.java

```diff
@@ -0,0 +1,226 @@
+                msgBuilder.append(" from file ");
+                msgBuilder.append(libFile.getAbsolutePath());
+                msgBuilder.append(". ");
+            }
+            msgBuilder.append("A confirmation message will follow if the loading is successful.");
+            s_hostLog.info(msgBuilder.toString());
+            if (useJavaLib) {
+                System.loadLibrary(fullLibName);
+            } else {
+                System.load(libFile.getAbsolutePath());
+            }
+            s_loadedLibs.add(name);
+            s_hostLog.info("Successfully loaded VoltDB native library " + fullLibName + ".");
+            return true;
+        } catch (Throwable e) {
+            if (s_hostLog.isDebugEnabled()) {
+                s_hostLog.debug("Error loading VoltDB JNI shared library", e);
+            }
+            if (useJavaLib) {
+                s_hostLog.info("Retry loading from file.");
+                return load(name, mustSucceed, false);
+            }
+            if (mustSucceed) {
+                msgBuilder.setLength(0);
+                msgBuilder.append("Failed to load shared library ").append(fullLibName).append(": ");
+                msgBuilder.append(e.getMessage()).append('\n');
+                msgBuilder.append("Library path: ").append(System.getProperty("java.library.path")).append('\n');
+                msgBuilder.append("The library may have failed to load because it cannot be found in your ");
+                msgBuilder.append("load library path, or because it is not compatible with the current platform.\n");
+                msgBuilder.append("VoltDB provides builds on our website for 64-bit OS X systems >= 10.6, ");
+                msgBuilder.append("and 64-bit Linux systems with kernels >= 2.6.18.");
+                if (e instanceof UnsatisfiedLinkError) {
+                    msgBuilder.append("\nOr the library may have failed to load because java.io.tmpdir should be set ");
+                    msgBuilder.append("to a different directory. Use VOLTDB_OPTS='-Djava.io.tmpdir=<dirpath>' to set it.");
+                }
+                VoltDB.crashLocalVoltDB(msgBuilder.toString(), false, null);
+            } else {
+                s_hostLog.info("Failed to load shared library " + fullLibName + "\nLibrary path: "
+                        + System.getProperty("java.library.path"));
+            }
+            return false;
+        }
+    }
+
+
+    /**
+     * Returns the native library file copied into a readable location.
+     * @param libname the library name.
+     * @return the native library file copied into a readable location.
+     */
+    private static File getNativeLibraryFile(String libname) {
+
+        // for now, arch is always x86_64
+        String pathFormat = "/org/voltdb/native/%s/x86_64";
+        String libPath = null;
+        if (System.getProperty("os.name").toLowerCase().contains("mac")) {
+            libPath = String.format(pathFormat, "Mac");
+        } else {
```

Comment:
In a more ideal world, I'd like to have
```suggestion
        } else if (System.getProperty("os.name").toLowerCase().contains("linux")) {
            libPath = String.format(pathFormat, "Linux");
        } else {
            throw new RuntimeError("Unsupported system: " + System.getProperty("os.name"));
         }
```